lumicks/pylake/kymo.py

```diff
@@ -237,6 +250,41 @@ def plot_with_time_histogram(self, color_channel, pixels_per_bin=1, hist_ratio=0
         ax_hist.set_ylabel("counts")
         ax_hist.set_title(self.name)
 
+    def downsampled_by(self, time_factor, reduce=np.sum):
+        """Return a copy of this Kymograph which is downsampled by `time_factor`
+
+        Parameters
+        ----------
+        time_factor : int
+            The number of pixels that will be averaged in time.
+        reduce : callable
+            The `numpy` function which is going to reduce multiple pixels into one.
+            The default is `np.sum`.
+        """
+        result = Kymo(self.name, self.file, self.start, self.stop, self._json)
+
+        def image_factory(_, channel):
+            data = self._image(channel)
+            return block_reduce(data, (1, time_factor), func=reduce)[
+                :, : data.shape[1] // time_factor
+            ]
+
+        def timestamp_factory(_):
+            raise AttributeError(
+                "Per pixel timestamps are no longer available after downsampling a kymograph since "
```

Comment:
I think this needs a hyphen since it's an adjective: "Per-pixel timestamps"

Reply:
I think you are correct :)